### PR TITLE
add disease name to flu only loincs in FHIR bundle

### LINKED ISSUE #6301: Add disease name to FHIR bundle for FLU only LOINCs
### Description
Add disease name to FHIR bundle for FLU only LOINCs

### PR DESCRIPTION
**Pull request by zdeveloper**
# BACKEND PULL REQUEST

## Related Issue

- resolves #6301

## Changes Proposed

- add disease name to flu only loincs in FHIR bundle that we dont have in our device table

## Review
I had to do a lot of manual loinc lookup for the disease type

- [x] Reviewer 1 please verify the following then check the box
```
"100973-7": "Flu-A"
"100974-5": "Flu-B"
"17015-9": "Flu-B"
"17016-7": "Flu-B"
"22096-2": "Flu-A"
"22825-4": "Flu-A"
"22827-0":"Flu-A"
"24015-0": "Flu A and B"
"31437-7": "Flu-A"
"31438-5": "Flu-A"
"31859-2": "Flu-A"
"31864-2": "Flu-B"
"33535-6": "Flu A and B"
"34487-9": "Flu-A"
"38381-0": "Flu-A"
"38382-8": "Flu-B"
"40982-1": "Flu-B"
"43874-7": "Flu-A"
"43895-2": "Flu-B"
"44558-5": "Flu-A"
"44563-5": "Flu-A"
"44564-3": "Flu-A"
"44567-6": "Flu A and B"
```
- [x] Reviewer 2 please verify the following then check the box
```
"44570-0": "Flu-B"
"44571-8": "Flu-B"
"44573-4": "Flu-B"
"44575-9": "Flu-B"
"44577-5": "Flu-B"
"46082-4": "Flu-A"
"46083-2": "Flu-B"
"48310-7": "Flu-A"
"48509-4": "Flu A and B"
"49521-8": "Flu-A"
"49523-4": "Flu-A"
"49524-2": "Flu-A"
"49531-7": "Flu-A"
"49535-8": "Flu-B"
"50697-2": "Flu-A"
"5229-0": "Flu-A"
"5230-8": "Flu-B"
"54243-1": "Flu RNA"
"55463-4": "Flu-A"
"55464-2": "Flu-A"
"55465-9": "Flu-A"
"5862-8": "Flu-A"
"5863-6": "Flu-A"
"5866-9": "Flu-B"
"59423-4": "Flu-A"
"62462-7": "Flu A and B"
"6435-2": "Flu A and B"
"6437-8": "Flu A and B"
"6438-6": "Flu A and B"
"68986-9": "Flu-A"
"68987-7": "Flu-A"
"72356-9": "Flu A and B"
"72366-8": "Flu A and B"
```
- [x] Reviewer 3 please verify the following then check the box
```
"74785-7": "Flu-B"
"74786-5": "Flu-B"
"74787-3": "Flu-B"
"76078-5": "Flu-A"
"76080-1": "Flu-B"
"77026-3": "Flu-A"
"77027-1": "Flu-A"
"77028-9": "Flu-A"
"7920-2": "Flu-A"
"7931-9": "Flu-B"
"80381-7": "Flu A and B"
"80382-5": "Flu-A"
"80383-3": "Flu-B"
"82166-0": "Flu-A"
"82167-8": "Flu-A"
"82168-6": "Flu-A"
"82169-4": "Flu-A"
"82170-2": "Flu-B"
"85476-0": "Flu A and B"
"85477-8": "Flu-A"
"85478-6": "Flu-B"
"92141-1": "Flu-B"
"92142-9": "Flu-A"
"92976-0": "Flu-B"
"92977-8": "Flu-A"
"9531-5": "Flu-A"
"9534-9": "Flu-B"
```

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhir.java [hunk 1]
```diff
@@ -441,11 +445,9 @@ private String getSpecimenTypeSnomed(String input) {
...
-    if (input.matches(ALPHABET_REGEX)) {
-      return input;
-    } else if (input.matches(SNOMED_REGEX)) {
-      return resultsUploaderCachingService.getSNOMEDToSpecimenTypeNameMap().get(input);
+  private String getSpecimenTypeDescription(String loinc) {
+    if (loinc != null) {
+      return resultsUploaderCachingService.getSNOMEDToSpecimenTypeNameMap().get(loinc);
     }
     return null;
   }
```
Comment (zdeveloper) [inline backend/src/main/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhir.java:453]: The change here doesn't alter the behavior, just better null handling and the removal of unnecessary regex matching

File: backend/src/main/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhir.java [hunk 2]
```diff
@@ -441,11 +445,9 @@ private String getSpecimenTypeSnomed(String input) {
...
 
-  private String getSpecimenTypeDescription(String input) {
-    if (input.matches(ALPHABET_REGEX)) {
-      return input;
-    } else if (input.matches(SNOMED_REGEX)) {
-      return resultsUploaderCachingService.getSNOMEDToSpecimenTypeNameMap().get(input);
+  private String getSpecimenTypeName(String loinc) {
```
Comment (zdeveloper) [inline backend/src/main/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhir.java:448]: Ooops this is not a loinc, its a SNOMED, I will have a follow up PR to fix it

File: backend/src/test/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhirTest.java [hunk 3]
```diff
@@ -343,6 +343,40 @@ void convertExistingCsv_matchesFhir_with_specimenType_LOINC() throws IOException
     assertThat(actualBundles).isEqualToIgnoringWhitespace(expectedBundleString);
   }
 
+  @Test
+  void convertExistingCsv_matchesFhir_with_flu_only() throws IOException {
```
Comment (nathancrtr) [inline backend/src/test/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhirTest.java:347]: Love the test 👏

**Review by nathancrtr — APPROVED** [review]
LGTM! Love how our FHIR convertor tests have been growing in-step with the service - this part of the app should have great coverage.

**Review by mpbrown — APPROVED** [review]
LGTM!

**Pull request merged**